src/main/java/co/cask/cdap/guides/workflow/PurchaseEventVerifier.java

```diff
@@ -22,21 +21,22 @@ public boolean apply(WorkflowContext workflowContext) {
       return false;
     }
 
-    Map<String, Map<String, Long>> hadoopCounters = token.getMapReduceCounters();
-    if (hadoopCounters == null) {
-      return false;
-    }
+    String taskCounterGroupName = "org.apache.hadoop.mapreduce.TaskCounter";
+    String mapInputRecordsCounterName = "MAP_INPUT_RECORDS";
 
-    Map<String, Long> taskCounter = hadoopCounters.get("org.apache.hadoop.mapreduce.TaskCounter");
+    Value mapInputRecords = token.get(taskCounterGroupName + "." + mapInputRecordsCounterName,
+                                      WorkflowToken.Scope.SYSTEM);
 
-    if (taskCounter.containsKey("MAP_INPUT_RECORDS")) {
-      long mapInputRecordNumber = taskCounter.get("MAP_INPUT_RECORDS");
-      long mapOutputRecordNumber = taskCounter.get("MAP_OUTPUT_RECORDS");
-      // Return true if at least 80% of the records were successfully parsed and emitted
-      // by previous map job
-      return (mapOutputRecordNumber >= (mapInputRecordNumber * 80/100));
+    String mapOutputRecordsCounterName = "MAP_OUTPUT_RECORDS";
```

Comment:
We can, however was explicitly using the variables here to highlight the counter group and name are separated by `.` character.